### ISSUE TITLE
Shell Grain on Windows

### ISSUE THREAD

**Issue by damon-atkins** [saltstack/salt#43506]
### Description of Issue/Question
```
C:\salt>salt-call grains.get shell
local:
    /bin/sh
C:\salt>salt-call --version
salt-call 2017.7.1 (Nitrogen)
```
Question why is it not C:\Windows\System32\cmd.exe or C:\Windows\SysWOW64\cmd.exe

**Comment by garethgreenaway** [comment]
@damon-atkins Looking at the code that generates the `shell` grain it looks for the environmental variable `SHELL` and if it doesn't find it then it returns `/bin/sh`.  Most likely that var doesn't exist under Windows. Should be trivial to detect a Windows system and return a different default or base the check on something else.

**Comment by damon-atkins** [comment]
I suspect it will need to happen in develop, as a patch to existing release might change the behaviour.

**Comment by damon-atkins** [comment]
Close as fix in PR #43552
